src/_pytest/compat.py

```diff
@@ -304,8 +305,8 @@ def _setup_collect_fakemodule():
 
     pytest.collect = ModuleType("pytest.collect")
     pytest.collect.__all__ = []  # used for setns
-    for attr in COLLECT_FAKEMODULE_ATTRIBUTES:
-        setattr(pytest.collect, attr, getattr(pytest, attr))
+    for attr_name in COLLECT_FAKEMODULE_ATTRIBUTES:
```

Comment:
can we remove that in pytest 5?

Reply:
We have this list of deprecations to remove in 5.1 (the deprecations will be errors in 5.0, removed in 5.1 as we did in 4.0/4.1).

Unfortunately it doesn't seem the `pytest.collect` has been deprecated at all, so unless I'm mistaken our only option is to deprecate it in `5.X` and remove in `6.X` I'm afraid. 😕

Reply:
Not sure how to implement this deprecation thought without using a `__getattr__` at module level, but that's 3.7+ only.

Reply:
```python
class MyModule(ModuleType):
    def __getattr__(...):
        ...

sys.modules[...] = ...
```

would work (it's essentially what py37 does)

Reply:
Yeah I had the exact same thought right after I posted my comment, but got side-tracked by #5452. 😛